src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -13,39 +14,25 @@ namespace System.Text.Json
     /// </summary>
     public sealed class JsonObject : JsonNode, IEnumerable<KeyValuePair<string, JsonNode>>
     {
-        internal readonly Dictionary<string, JsonNode> _dictionary;
-        private readonly DuplicatePropertyNameHandling _duplicatePropertyNameHandling;
+        internal readonly Dictionary<string, JsonObjectProperty> _dictionary;
+        internal JsonObjectProperty _first;
+        internal JsonObjectProperty _last;
+        internal int _version;
 
         /// <summary>
         ///   Initializes a new instance of the <see cref="JsonObject"/> class representing the empty object.
         /// </summary>
-        /// <param name="duplicatePropertyNameHandling">Specifies the way of handling duplicate property names.</param>
-        /// <exception cref="ArgumentException">
-        ///   Provided manner of handling duplicates does not exist.
-        /// </exception>
-        public JsonObject(DuplicatePropertyNameHandling duplicatePropertyNameHandling = DuplicatePropertyNameHandling.Replace)
+        public JsonObject()
         {
-            if ((uint)duplicatePropertyNameHandling > (uint)DuplicatePropertyNameHandling.Error)
-            {
-                throw new ArgumentOutOfRangeException(SR.InvalidDuplicatePropertyNameHandling);
-            }
-
-            _dictionary = new Dictionary<string, JsonNode>();
-            _duplicatePropertyNameHandling = duplicatePropertyNameHandling;
+            _dictionary = new Dictionary<string, JsonObjectProperty>();
```

Comment:
Should StringComparison.Ordinal be passed to the dictionary? I think it is the default but being explicit can't hurt

Reply:
Ok, I will check it. I am now working on adding `StringComparison`.

Reply:
The default comparer for dictionary keys is the equals method for key - string, which performs an ordinal (case-sensitive and culture-insensitive) comparison, so this should be what we want.